jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/OdeMessageFrameDataCreatorHelper.java

```diff
@@ -0,0 +1,71 @@
+package us.dot.its.jpo.ode.coder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import us.dot.its.jpo.asn.j2735.r2024.MessageFrame.MessageFrame;
+import us.dot.its.jpo.ode.model.OdeMessageFrameData;
+import us.dot.its.jpo.ode.model.OdeMessageFrameMetadata;
+import us.dot.its.jpo.ode.model.OdeMessageFramePayload;
+import us.dot.its.jpo.ode.model.OdeMsgMetadata;
+import us.dot.its.jpo.ode.model.ReceivedMessageDetails;
+import us.dot.its.jpo.ode.model.RxSource;
+
+/**
+ * Helper class for creating OdeMessageFrameData objects from consumed data.
+ */
+@Slf4j
+public class OdeMessageFrameDataCreatorHelper {
+
+  private OdeMessageFrameDataCreatorHelper() {
+    throw new UnsupportedOperationException("Utility class should not be instantiated");
+  }
+
+  /**
+   * Creates an OdeMessageFrameData object from consumed XML data.
+   *
+   * @param consumedData The XML data string to be processed
+   * @param simpleObjectMapper The ObjectMapper for simple JSON operations
+   * @param simpleXmlMapper The XmlMapper for XML operations
+   * @return OdeMessageFrameData object containing the processed data
+   * @throws JsonProcessingException if there is an error processing the JSON data
+   */
+  public static OdeMessageFrameData createOdeMessageFrameData(String consumedData, 
+      ObjectMapper simpleObjectMapper, XmlMapper simpleXmlMapper) throws JsonProcessingException {
+    ObjectNode consumed = simpleXmlMapper.readValue(consumedData, ObjectNode.class);
+    JsonNode metadataNode = consumed.findValue(OdeMsgMetadata.METADATA_STRING);
+    if (metadataNode instanceof ObjectNode object) {
+      object.remove(OdeMsgMetadata.ENCODINGS_STRING);
+
+      // Header file does not have a location and use predefined set required RxSource
+      ReceivedMessageDetails receivedMessageDetails = new ReceivedMessageDetails();
+      receivedMessageDetails.setRxSource(RxSource.NA);
+
+      JsonNode jsonNode;
+      try {
+        jsonNode = simpleObjectMapper.readTree(receivedMessageDetails.toJson());
+        object.set(OdeMsgMetadata.RECEIVEDMSGDETAILS_STRING, jsonNode);
+      } catch (IOException e) {
+        log.error("Failed to parse receivedMessageDetails", e);
+        throw new RuntimeException(e);
+      }
+    }
+
+    OdeMessageFrameMetadata metadata =
+        simpleXmlMapper.convertValue(metadataNode, OdeMessageFrameMetadata.class);
+
```

Comment:
suggestion(non-blocking): since we aren't deserializing any data into the ReceivedMessageDetails object, we can remove a good chunk of code (including a try-catch block) and set the field directly on the OdeMessageFrameMetadata object instead. This improves the code clarity and removes the possibility of the RuntimeException.
```suggestion
    }

    OdeMessageFrameMetadata metadata =
        simpleXmlMapper.convertValue(metadataNode, OdeMessageFrameMetadata.class);
    OdeMessageFrameMetadata metadata =
        simpleXmlMapper.convertValue(metadataNode, OdeMessageFrameMetadata.class);
    // Header file does not have a location and use predefined set required RxSource
    metadata.getReceivedMessageDetails().setRxSource(RxSource.NA);
```

Reply:
addressed